packages/ui/src/proposals/components/ProposalStages.tsx

```diff
@@ -3,44 +3,73 @@ import React, { useMemo } from 'react'
 import { ControlProps } from '@/common/components/forms'
 import { CheckboxIcon, CrossIcon } from '@/common/components/icons'
 import { TabContainer, TabsContainer } from '@/common/components/Tabs'
+import { Tooltip } from '@/common/components/Tooltip'
 import { last, repeat } from '@/common/utils'
 import { ProposalStatus, ProposalStatusUpdates } from '@/proposals/types'
 
 export interface ProposalStagesProps extends ControlProps<number> {
-  status: ProposalStatus
+  roundStatus: ProposalStatus
   updates: ProposalStatusUpdates[]
-  constitutionality?: number | string
+  constitutionality?: number
 }
 
-export const ProposalStages = ({ status, updates, constitutionality = '-', value, onChange }: ProposalStagesProps) => {
-  const rounds = useMemo(() => {
+type RoundState = 'approved' | 'rejected' | 'deciding' | 'disabled'
+const iconMap = { approved: <CheckboxIcon />, rejected: <CrossIcon />, deciding: undefined, disabled: undefined }
+
+export const ProposalStages = ({ roundStatus, updates, constitutionality, value, onChange }: ProposalStagesProps) => {
+  const rounds: RoundState[] = useMemo(() => {
     const decidingCount = updates.filter(({ status }) => status === 'deciding').length
+
     const lastUpdate = last(updates).status
-    const onGoing = lastUpdate === status
+    const onGoing = lastUpdate === roundStatus
     const approved = lastUpdate === 'gracing'
     const rejected = !onGoing && !approved
-    const isDeciding = onGoing && status === 'deciding'
-    const isDormant = onGoing && status === 'dormant'
-    return [
-      ...repeat((round) => ({ icon: <CheckboxIcon />, onClick: () => onChange(round) }), decidingCount - 1),
-      {
-        icon: !isDeciding && (rejected ? <CrossIcon /> : <CheckboxIcon />),
-        onClick: () => onChange(decidingCount - 1),
-      },
-      ...(isDormant ? [{ icon: false, onClick: undefined }] : []),
-    ]
-  }, [updates.length, status])
+    const isDormant = onGoing && roundStatus === 'dormant'
+
+    return repeat((round) => {
+      if (round < decidingCount) {
+        return 'approved'
+      } else if (round > decidingCount) {
```

Comment:
It should be `decidingCount - 1`:
```suggestion
      if (round < decidingCount - 1) {
        return 'approved'
      } else if (round > decidingCount - 1) {
```

Reply:
Ok. I will do.